### PR TITLE
Fix/incorrect build dependencies

### PR DESCRIPTION
**Pull request by patrick-tolosa**
Fix for incorrect build output.

There were three issues with the existing project:

1. Browser field would be incorrectly used by Webpack, pointing to src/index.js - a file which is not transpiled to ES5 and thus breaks builds due to syntax.
2. Main field incorrectly points to to common.js instead of umd, making the main entry point invalid for bundlers
3. The build output would reference core-js, specifically core-js@2, since core-js is not a dependency of this project, it would use the parent's core-js. if the user uses core-js@3, the project would not not be consumable. 

This PR fixes all three.

### CODE REVIEW

**Comment by asisayag2** [comment]
@mayashavin , can you pls review?

File: babel.config.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
-  presets: ["@vue/app"]
+  presets: [["@vue/app",{useBuiltIns: false}]],
```
Comment (mayashavin) [inline babel.config.js]: by default, `useBuiltIns` is `false`, what does this change help in general? 
https://babeljs.io/docs/en/babel-preset-env
Don't we need to add polyfill?

Comment (patrick-tolosa) [reply babel.config.js]: The default @vue/app preset we use (that's part of vue-CLI I guess) has this configuration enabled, so what we're doing here is forcing it to be false.

Screen Shot 2020-02-27 at 11 40 54

Two other choices are to include core-js with our desired version, Or to force a peer-dependency of the specified version, the latter is a no-go if we want to a broad spectrum of builds

Comment (patrick-tolosa) [reply babel.config.js]: @mayashavin  - I added core-js@2.6.9 as a dependency and removed the builtins, that should solve our problems while keeping this library self-contained and not dependent on the consuming project

**Pull request merged**